src/Shared/FileUtilities.cs

```diff
@@ -243,6 +243,58 @@ internal static string EnsureNoTrailingSlash(string path)
             return path;
         }
 
+        /// <summary>
+        /// Ensures the path is enclosed within single quotes.
+        /// </summary>
+        /// <param name="path"></param>
+        /// <returns></returns>
+        internal static string EnsureSingleQuotes(string path)
+        {
+            return EnsureQuotes(path);
+        }
+
+        /// <summary>
+        /// Ensures the path is enclosed within double quotes.
+        /// </summary>
+        /// <param name="path"></param>
+        /// <returns></returns>
+        internal static string EnsureDoubleQuotes(string path)
+        {
+            return EnsureQuotes(path, isSingleQuote: false);
+        }
+
+        /// <summary>
+        /// Ensures the path is enclosed within quotes.
+        /// </summary>
+        /// <param name="path"></param>
+        /// <param name="isSingleQuote">indicates if single or double quotes should be used</param>
+        /// <returns></returns>
+        internal static string EnsureQuotes(string path, bool isSingleQuote = true)
+        {
+            path = FixFilePath(path);
+
+            const char singleQuote = '\'';
```

Comment:
We can avoid this char if we use `MSBuildConstants.SingleQuoteChar[0]`. Though it apparently doesn't have a `doubleQuoteChar`? We can add one in `Shared/Constants.cs` in this PR and use it in Evaluator since it creates a  `private static readonly char[] s_doubleQuoteChar = { '"' };`

If you'd rather avoid that, I can create an `up-for-grabs` issue for it.

Reply:
The fields in `MSBuildConstants` are char arrays. The comment says `// One-time allocations to avoid implicit allocations for Split(), Trim().` which makes sense. There is a memory and performance cost to recreating a char array on every use. But that isn't true for a char literal. I don't think that it makes sense to make a change to use `MSBuildConstants.SingleQuoteChar[0]`. Additionally there are char literals all over the codebase.

Reply:
Yes, the code in `ExecMSBuild`:
```
    msbuildParameters = "\"" + pathToMsBuildExe + "\"" + " " + msbuildParameters;
```
could be changed to:
```
    msbuildParameters = FileUtilities.EnsureDoubleQuotes(pathToMsBuildExe) + " " + msbuildParameters;
```

I'l add that change.

Reply:
Done